lib/plugins/aws/package/index.js

```diff
@@ -76,6 +81,9 @@ class AwsPackage {
       'aws:package:finalize:mergeCustomProviderResources': () => BbPromise.bind(this)
         .then(this.mergeCustomProviderResources),
 
+      'after:aws:package:finalize:mergeCustomProviderResources': () => BbPromise.bind(this)
```

Comment:
`before:aws:deploy:updateStack` (see above) would be the right and only place for a check.